modules/indexer/code/queue.go

```diff
@@ -22,8 +22,6 @@ type repoIndexerOperation struct {
 var repoIndexerOperationQueue chan repoIndexerOperation
 
 func processRepoIndexerOperationQueue(indexer Indexer) {
-	defer indexer.Close()
```

Comment:
Once the indexer is running we should close it at Terminate not Shutdown. Things running in the ShutdownContext and up to HammerTime might still need to send data to the repo indexer.

If the indexer fails to be set up then we should close ASAP as we want gitea to stop ASAP.